test/cql-pytest/test_secondary_index.py

```diff
@@ -21,6 +21,7 @@
 import pytest
 from cassandra.protocol import SyntaxException, AlreadyExists, InvalidRequest, ConfigurationException, ReadFailure
 from cassandra.query import SimpleStatement
+from cassandra_tests.porting import assert_rows
```

Comment:
I didn't think we would use cassandra_tests.porting outside the cassandra_tests directory (tests translated from Cassandra), but I guess we can...

Reply:
Arguably, `assert_rows` is universal goodness that should be available outside `cassandra_tests.porting`.  But that's for another PR.

Reply:
Maybe, although in most simple cases not involving complex frozen types, assert_rows is nothing more than "assert list(cql.query(...)) = ...".

Reply:
IOW, it's strictly better than `assert list(...)=...`.  Universal goodness. :)  Why is it in cassandra_tests.porting?

Reply:
It's in cassandra_tests.porting because the Cassandra tests in Java used a functionc called AssertRows and I wanted to have a similar function in Python to make conversion easier.
It's fine to import cassandra_tests.porting for now. Later if we think such functions are useful we can move them to util.py which is a more logical place for them. I just want to point out that in general, such assert functions are rarely needed in pytest, which has a very powerful assert mechanism where you just do "assert ... == ..." and it not only checks for equality, but also tries hard to tell you when there is inequality, what exactly didn't match. So functions like assert_rows() are much less important in pytest than they are in the Java or C++ unit tests. But I am not completely against them, if other people find them useful.